### ISSUE TITLE
Filter user attestation by source platform

### ISSUE THREAD

**Issue by divine-comedian** [Giveth/DeVouch-BE#24]

**Comment by aminlatifi** [comment]
Sample query 
```graphql
{
  projectAttestations(where: {attestorOrganisation: {attestor: {id_eq: "0x826976d7c600d45fb8287ca1d7c76fc8eb732030"}}, project: {source_eq: "giveth"}}) {
    attestTimestamp
    comment
    vouch
    project {
      source
    }
    attestorOrganisation {
      organisation {
        name
      }
    }
  }
}

```
Sample Response:
```json
{
  "data": {
    "projectAttestations": [
      {
        "attestTimestamp": "2024-05-09T17:36:36.000000Z",
        "comment": "this is awesome",
        "vouch": true,
        "project": {
          "source": "giveth"
        },
        "attestorOrganisation": {
          "organisation": {
            "name": "Giveth Verification Team"
          }
        }
      },
      {
        "attestTimestamp": "2024-05-10T12:24:36.000000Z",
        "comment": "cool project! good luck",
        "vouch": true,
        "project": {
          "source": "giveth"
        },
        "attestorOrganisation": {
          "organisation": {
            "name": "Trace"
          }
        }
      },
      {
        "attestTimestamp": "2024-05-12T16:26:36.000000Z",
        "comment": "Does it show up now?",
        "vouch": true,
        "project": {
          "source": "giveth"
        },
        "attestorOrganisation": {
          "organisation": {
            "name": "Trace"
          }
        }
      }
    ]
  }
}
```

**Comment by maryjaf** [comment]
> * [ ]  filter by gitcoin source works
> * [ ]  filter by rpgf source works
> * [ ]  when filtering does not show attestations made to projects on other source platforms

- [ ] Now I can only test "source: giveth" and there is no project on gitcoin or RPGF @divine-comedian

**Comment by maryjaf** [comment]
> * [ ]  - Title of project attested to
> * [ ]  link/slug to project details page

- [ ] 2-
Title is "null" and I haven't seen "link/slug" field in response @aminlatifi

**Comment by maryjaf** [comment]
- [ ] 3-
The revoked attestianio also should be shown in the response ? @aminlatifi 
https://sepolia.easscan.org/attestation/view/0x0ccc0236015d469dd34be5d54a95a22559f71af749ace58f1d9a399864cb3fe5
This revoked attestation is shown in the response, is this expected ?

**Comment by aminlatifi** [comment]
> * [ ]  3-
>   The revoked attestianio also should be shown in the response ? @aminlatifi
>   https://sepolia.easscan.org/attestation/view/0x0ccc0236015d469dd34be5d54a95a22559f71af749ace58f1d9a399864cb3fe5
>   This revoked attestation is shown in the response, is this expected ?

They can be shown, but their `revoked` field must be true. Important point is they must not be counted in project stats like totalVouch, totalFlag, totalAttestation and associated organisation counters.

**Comment by aminlatifi** [comment]
> @aminlatifi we don't need to show revoked attestations on the response, we consider them as "deleted" once the user revokes them, is it possible to remove them from the response? it will simplify the logic of the FE

Yes, it's possible, we left it as an option. So, we have defined a `revoked` field in the attestations, should we remove it as well?!

**Comment by divine-comedian** [comment]
@MohammadPCh @aminlatifi  - we'll need to know when/how we can make attestations for other source platforms in order to test this one thoroughly

**Comment by aminlatifi** [comment]
> @MohammadPCh @aminlatifi - we'll need to know when/how we can make attestations for other source platforms in order to test this one thoroughly

@MohammadPCh has implemented all importing... I think the challenge is figuring out how to fetch valid projects from the source platform API. The rest is designed well by Cherik to support various response types. Am I right @MohammadPCh ?

**Comment by divine-comedian** [comment]
@aminlatifi I followed up on this one based on MJ's testing - everything looks good. I can see when I revoke an attestations it reflect seamlessly in the front-end so I feel satisfied with its current state. Moving this to done
